### ISSUE TITLE
array-bracket-spacing: miss-handles type annotations

### ISSUE THREAD

**Issue by zaygraveyard** [eslint/eslint#4753]
### test.js
```js
/*eslint array-bracket-spacing: [2, "never"] */

function fnArray([a, b]: Array): void {
  // nothing important
}
```

### What versions I'm using
eslint@1.10.3
babel-eslint@4.1.6

### What I did
```sh
$eslint --parser babel-eslint test.js
```

### What I expected
No errors

### What actually happened
```
/tmp/test.js
  3:26  error  There should be no space before 'Array'  array-bracket-spacing

✖ 1 problem (1 error, 0 warnings)
```

### My Observations

The `ArrayPattern` node of the AST tree contains the type annotation.

```js
function fnArray([a, b]: Array): void {
//               ^-----------^
//                ArrayPattern
```

The `array-bracket-spacing` rule checks the last and before last tokens of the AST node (ie. last is `Array` an before last is `:`).

Since the between the before last token and last token their is a space and the expected is no space, an error is reported.

**Comment by zaygraveyard** [comment]
Working on it

**Comment by ilyavolodin** [comment]
Closing, as we do not support flow as part of the core ruleset. You might want to open an issue for [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel)
